internal/services/appservice/linux_function_app_slot_resource.go

```diff
@@ -655,6 +655,13 @@ func (r LinuxFunctionAppSlotResource) Update() sdk.ResourceFunc {
 				return fmt.Errorf("reading Linux %s: %v", id, err)
 			}
 
+			_, planSKU, err := helpers.ServicePlanInfoForApp(ctx, metadata, *id)
+			if err != nil {
+				return err
+			}
+
+			sendContentSettings := !helpers.PlanIsElastic(*planSKU)
```

Comment:
should helpers.PlanIsElastic take a nil pointer so avoid all the derefs (and then always check for nil) ?